src/traces/bar/plot.js

```diff
@@ -215,27 +216,140 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts, makeOnCompleteCallback)
                 (v > vc ? Math.ceil(v) : Math.floor(v));
             }
 
+            var op = Color.opacity(mc);
+            var fixpx = (op < 1 || lw > 0.01) ? roundWithLine : expandToVisible;
             if(!gd._context.staticPlot) {
                 // if bars are not fully opaque or they have a line
                 // around them, round to integer pixels, mainly for
                 // safari so we prevent overlaps from its expansive
                 // pixelation. if the bars ARE fully opaque and have
                 // no line, expand to a full pixel to make sure we
                 // can see them
-
-                var op = Color.opacity(mc);
-                var fixpx = (op < 1 || lw > 0.01) ? roundWithLine : expandToVisible;
-
                 x0 = fixpx(x0, x1, isHorizontal);
                 x1 = fixpx(x1, x0, isHorizontal);
                 y0 = fixpx(y0, y1, !isHorizontal);
                 y1 = fixpx(y1, y0, !isHorizontal);
             }
 
+            // Function to convert from size axis values to pixels
+            var c2p = isHorizontal ? xa.c2p : ya.c2p;
+
+            // Calculate corner radius of bar in pixels
+            function calcCornerRadius(radiusParam) {
+                var barWidth = isHorizontal ? Math.abs(y1 - y0) : Math.abs(x1 - x0);
+                var barLength = isHorizontal ? Math.abs(x1 - x0) : Math.abs(y1 - y0);
+                var stackedBarTotalLength = fixpx(Math.abs(
+                    di.s > 0 ? c2p(di._sMax, true) - c2p(0, true) : c2p(di._sMin, true) - c2p(0, true)
+                ));
+                var maxRadius = di.hasB ? Math.min(barWidth / 2, barLength / 2) : Math.min(barWidth / 2, stackedBarTotalLength);
+                var rPx;
+                if(!radiusParam) {
+                    return 0;
+                } else if(typeof radiusParam === 'string') {
```

Comment:
We shouldn't get into this if statement, if the value is presented in a string but not having a percentage at the end.
Also we may replace `parseFloat` by unary plus operator to get `NaN` when the remaining string is not a number.
Something like:
```js
                } else if(typeof radiusParam === 'string' && radiusParam.slice(-1) === '%') {
                    // If radius is given as a percentage string, convert to number of pixels
                    var rPercent = Math.min(50, +radiusParam.slice(0, -1));
```

Reply:
Better to use `isNumeric` to find if the value is a number, it's our standard method of finding either number type or numeric string:
```js
if(!radiusParam) {
    return 0;
else if(isNumeric(radiusParam)) {
    rPx = +radiusParam;
else {
    var rPercent = ...
}
```

Reply:
If `cornerradius` value is invalid (e.g. string not ending in %) should we log an error message or just ignore?

Reply:
Ignore if it’s invalid. Plotlyjs has a separate validation utility that will alert you if any attributes you provide were not accepted, but if you don’t explicitly call that our convention is to silently ignore invalid items.